src/pages/library/datasetBuilder/CohortEditor.ts

```diff
@@ -176,9 +178,18 @@ const createDefaultRangeCriteria = (rangeOption: ProgramDataRangeOption): Progra
   };
 };
 
-type CriteriaOption = DomainOption | ProgramDataRangeOption | ProgramDataListOption;
+const addKindToDomainOption = (domainOption: DomainOption): DomainOptionWithKind => ({
+  ...domainOption,
+  kind: 'domain',
```

Comment:
Why are `ProgramDataListOption` etc populated with the correct `kind` value but `DomainOption` isn't?

If `DomainOption` need to be modified before use, could it be done in the ajax API layer so any downstream code doesn't need to copy that step?

Reply:
`ProgramDataListOption` has the kind because there is a difference between types within one group. Because everything in `snapshotBuilderSettings.domainOptions` has the same kind, it is derivable and therefore should be omitted from the API response. It is important for the UI to distinguish though, because the UI has to render them in a single group, and thus they need to be treated the same (even if they are separate from the API standpoint).

I'll move the modification to be at the ajax API layer.

Reply:
Hmmm... actually I don't like putting it at the ajax level. Maybe we can discuss it offline? I do agree we don't want to add it multiple times, but I think right now its just being added in one place, where we need to cross display it with programDataOptions